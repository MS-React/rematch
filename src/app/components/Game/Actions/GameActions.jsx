import React from 'react';
import PropTypes from 'prop-types';

const GameActions = ({
  setResult, confirmProof, resumePause, playing,
}) => (
  <div key="game-user-actions" className="game-actions">
    <input type="text" onChange={setResult} />
    <button type="submit" className="action-confirmProof" onClick={confirmProof}>
      Confirm
    </button>
    <button type="submit" onClick={resumePause}>
      {playing && 'Pause'}
      {!playing && 'Resume'}
    </button>
  </div>
);

GameActions.propTypes = {
  setResult: PropTypes.func.isRequired,
  confirmProof: PropTypes.func.isRequired,
  resumePause: PropTypes.func.isRequired,
  playing: PropTypes.bool,
};

GameActions.defaultProps = {
  playing: false,
};

export default GameActions;
