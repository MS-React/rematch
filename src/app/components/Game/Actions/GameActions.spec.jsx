import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import GameActions from './GameActions';

describe('GameActions', () => {
  let wrapper;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      setResult: jest.fn(),
      confirmProof: jest.fn(),
      resumePause: jest.fn(),
      playing: false
    };
    wrapper = shallow(<GameActions {...defaultProps} />);
  });

  it('should show game actions to confirm the proof and pause/resume with pause as first to the user', () => {
    expect(renderer.create(wrapper).toJSON())
      .toMatchSnapshot();
  });
});
