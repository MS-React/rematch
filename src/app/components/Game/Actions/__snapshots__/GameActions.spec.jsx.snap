// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameActions should show game actions to confirm the proof and pause/resume with pause as first to the user 1`] = `
<div
  className="game-actions"
>
  <input
    onChange={[MockFunction]}
    type="text"
  />
  <button
    className="action-confirmProof"
    onClick={[MockFunction]}
    type="submit"
  >
    Confirm
  </button>
  <button
    onClick={[MockFunction]}
    type="submit"
  >
    Resume
  </button>
</div>
`;
